airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/NoOpTyperDeduperWithV1V2Migrations.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.base.destination.typing_deduping;
+
+import static io.airbyte.cdk.integrations.base.IntegrationRunner.TYPE_AND_DEDUPE_THREAD_NAME;
+import static io.airbyte.integrations.base.destination.typing_deduping.FutureUtils.countOfTypingDedupingThreads;
+import static io.airbyte.integrations.base.destination.typing_deduping.FutureUtils.reduceExceptions;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+
+/**
+ * This is a NoOp implementation which skips and Typing and Deduping operations and does not emit
+ * the final tables. However, this implementation still performs V1->V2 migrations and V2
+ * json->string migrations in the raw tables.
+ */
+@Slf4j
+public class NoOpTyperDeduperWithV1V2Migrations<DialectTableDefinition> implements TyperDeduper {
```

Comment:
maybe makes sense to add a `doTypingDeduping` flag to the default typerdeduper instead of adding a separate class? (I'm legitimately not sure which approach is better)

Reply:
(Kinda habit of Open/Closed principle 😅) Since I already see `TyperDeduper` as interface and callers only rely on the exposed method contracts, seemed like having a parallel impl with exactly what it does and not too much conditional branching seemed better (although i could blindly put `if(!doTypeDedupe) return;` but preferred polymorphism over conditional behavior of a class.

Reply:
weird tradeoff. If we modify how migrations work, we'll need to modify two classes... but if we modify how T+D works, there's less risk of breaking this

:shrug: let's go with the two-class structure. If it feels bad we can always change it 🚛